caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -842,11 +878,14 @@ def encode(
         error.type_check_all("<NLP82314994E>", str, sentences=list_of_sentences)
 
         if device is None:
-            device = self._target_device
+            device = self.device
 
         self.to(device)
 
         all_embeddings = []
+
+        # Sort sentences according to length, from longest to shortest
+        # OOM errors then occurs at start of encoding
```

Comment:
It is a comment I copied from the PR from `SentenceTransformer` where they explain why the sort order is used:
https://github.com/UKPLab/sentence-transformers/commit/ab7387c8956af840f55354f8981ba3fb2bfb6954

Since we are copying their code, I thought it was useful to at least partially explain why the sorting is done.